lib/rules/no-java-constructor.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * @fileoverview Rule to warn about declaring a method with the same name of the class. The
+ * developer was likely writing Java recently and meant to write `constructor`.
+ * @author Michael Bolin
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow method names that match the class name",
+            category: "Possible Errors",
+            recommended: true
+        },
+
+        // TODO(mbolin): Find out what is supposed to go here.
+        schema: {
```

Comment:
`schema` is used to configure options that a rule can accept.

Since this rule doesn't accept any options, you would probably want to set `schema` to an empty array.